packages/cli/src/commands/test.js

```diff
@@ -62,22 +57,51 @@ export const builder = (yargs) => {
 }
 
 export const handler = async ({
-  side,
+  filter: filterParams = [],
   watch = true,
   collectCoverage = false,
+  ...others
 }) => {
+  const forwardJestFlags = Object.keys(others).flatMap((flagName) => {
+    if (['watch', 'collect-coverage', '$0', '_'].includes(flagName)) {
+      // filter out flags meant for the rw test command only
+      return []
+    } else {
+      // and forward on the other flags
+      return [
+        flagName.length > 1 ? `--${flagName}` : `-${flagName}`,
+        others[flagName],
+      ]
+    }
+  })
+
   const { cache: CACHE_DIR } = getPaths()
-  const sides = [].concat(side).filter(Boolean)
-  const args = [
-    '--passWithNoTests',
+
+  // Only the side params
+  const sides = filterParams.filter((filterString) =>
+    getProject().sides.includes(filterString)
+  )
+
+  // All the other params, apart from sides
+  const jestFilterArgs = [
+    ...filterParams.filter(
+      (filterString) => !getProject().sides.includes(filterString)
+    ),
+  ]
+
+  const jestArgs = [
+    ...jestFilterArgs,
+    ...forwardJestFlags,
     collectCoverage && '--collectCoverage',
+    '--passWithNoTests',
+    ...jestFilterArgs,
   ].filter(Boolean)
 
   // If the user wants to watch, set the proper watch flag based on what kind of repo this is
   // because of https://github.com/facebook/create-react-app/issues/5210
   if (watch && !process.env.CI && !collectCoverage) {
     const hasSourceControl = isInGitRepository() || isInMercurialRepository()
-    args.push(hasSourceControl ? '--watch' : '--watchAll')
+    jestArgs.push(hasSourceControl ? '--watch' : '--watchAll')
   }
```

Comment:
Unrelated to this PR but I've always wondered about this... I've always disliked this piece of code. I think now that we're forwarding things to jest we shouldn't interfere with how it works.

If you attempt to run `yarn jest --watch` and you do not have version controll you'll get a warning that you need to run `yarn jest --watchAll`

I would much rather remove this code and just stick to what jest does. Honestly who builds software without version control, writes tests, and then expects different behaviour from the test commands?

🤷‍♂️

Reply:
100% agreed

Reply:
Sweet, I've opened an issue here: https://github.com/redwoodjs/redwood/issues/2708